recipes-core/systemd/systemd_%.bbappend

```diff
@@ -84,7 +84,11 @@ do_install:append:phyboard-polis-imx8mm-4() {
 
 # adapt tauri-l systemd-networkd-wait-online.service state
 do_install:append:phygate-tauri-l-imx8mm-2() {
-    sed -i -e 's/^ExecStart=\(.*\)/ExecStart=\1 --any --interface=${OMNECT_ETH0} --interface=${OMNECT_ETH1}/' ${D}${systemd_system_unitdir}/systemd-networkd-wait-online.service
+    if [ "${OMNECT_WWAN0}" ]; then
```

Comment:
what I had in mind with intentionally leaving OMNECT_WWAN0 undefined is the possibility to generate images lacking cellular access capability.
basically, this could be applied to WLAN functionality, too, yes, but on the other hand WLAN is much more common than LTE imho, and I don't think that someone would chose devices with (integrated) WLAN interfaces without wanting to use them.
all in all, it wouldn't do any harm do treat WLAN the same way, and maybe we should really do it for the sake of uniform handling; I would definitely not vote against that :-)
the whole discussion is mainly about device variant handling and how to handle them: while we definitely want to keep the number of images as low as possible, we could be forced to image variants in case support of full modularity no longer fits in the maximum acceptable rootfs size.
as long as we don't need to think about that, it doesn't matter which way WLAN and WWAN devices are handled here, but once we have to think about it, we would be prepared a bit more already for taking out that functionality.

Reply:
@JanZachmann we need a refinement